src/org/labkey/test/components/ui/grids/ResponsiveGrid.java

```diff
@@ -193,6 +194,30 @@ public T filterBooleanColumn(String fieldLabel, boolean value)
         return _this;
     }
 
+    /**
+     *
+     * @param index
+     * @param fieldLabel
+     * @param operator
+     * @param dateString1
+     * @param dateString2
+     * @return
+     */
+    public T filterDateColumn(int index, String fieldLabel, Filter.Operator operator, String dateString1, String dateString2)
+    {
+        T _this = getThis();
+        doAndWaitForUpdate(() -> {
+            clickColumnMenuItem(fieldLabel, "Filter...", false);
+            GridFilterModal filterModal = new GridFilterModal(getDriver(), this);
+            filterModal.selectExpressionTab().setFilterType(index, operator);
+            Input.Input(Locator.input("field-value-date-0"), getDriver()).waitFor().set(dateString1);
+            if (dateString2 != null)
+                Input.Input(Locator.input("field-value-date-1"), getDriver()).waitFor().set(dateString2);
+            filterModal.confirm();
+        });
+        return _this;
```

Comment:
Why do these filter methods stash `_this` and return it later? Seems unnecessary.
```suggestion
        return getThis();
```